src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -1148,6 +1152,44 @@ private fun checkDuplicates(
     message.addToHolder(holder)
 }
 
+private fun checkConstGenerics(holder: RsAnnotationHolder, constParameter: RsConstParameter) {
+    if (CONST_GENERICS.availability(constParameter) == AVAILABLE) return
+
+    val feature = when (constParameter.typeReference?.type) {
+        is TyInteger, is TyBool, is TyChar -> {
+            val current = constParameter.cargoProject?.rustcInfo?.version?.semver
+            val since = MIN_CONST_GENERICS.since
+            if (current != null && since != null && current >= since) MIN_CONST_GENERICS else CONST_GENERICS
+        }
+        else -> CONST_GENERICS
+    }
+
+    val fix = if (feature === CONST_GENERICS) {
+        createReplaceFeatureFix(constParameter, CONST_GENERICS.name, MIN_CONST_GENERICS.name)
+    } else {
+        null
+    }
+
+    feature.check(holder, constParameter, "const generics", *listOfNotNull(fix).toTypedArray())
+}
+
+private fun createReplaceFeatureFix(element: RsElement, newFixName: String, oldFixName: String): LocalQuickFix? {
+    val project = element.project
+    val crateRoot = element.crateRoot
+    val oldAttr = RsFeatureIndex.getFeatureAttributes(project, oldFixName)
+        .find { it.crateRoot == crateRoot }
+        ?: return null
+    return object : LocalQuickFixAndIntentionActionOnPsiElement(element) {
```

Comment:
Can we write test(s) for this quick-fix?

Reply:
👍